daft/dataframe/dataframe.py

```diff
@@ -280,13 +280,61 @@ def from_pydict(cls, data: Dict[str, InputListType]) -> "DataFrame":
             )
 
         data_vpartition = Table.from_pydict(data)
-        result_pset = LocalPartitionSet({0: data_vpartition})
+        return cls._from_table(data_vpartition)
+
+    @classmethod
+    @DataframePublicAPI
+    def from_arrow(cls, data: "pa.Table") -> "DataFrame":
```

Comment:
these don't have to be strings for the types right?

Reply:
pyarrow isn't imported eagerly for this module, it's only imported in the global scope for type checking, which then requires these to be string-typed:
```
daft/dataframe/dataframe.py:287: in DataFrame
    def from_arrow(cls, data: Union[pa.Table, List[pa.Table]]) -> "DataFrame":
E   NameError: name 'pa' is not defined
```